enemy/enemy_spawn.gd

```diff
@@ -12,20 +12,19 @@ const Mushroom = preload("res://enemy/mushroom/mushroom.tscn")
 const ENEMIES: Dictionary = {1: [FlyingEye], 2: [Goblin], 4: [Skeleton], 6: [Mushroom]}
 
 
-func spawn_wave(wave: Array) -> void:
-	for i in wave.size():
-		var item = wave[i]
-		var enemy = _spawn(item.score)
-		var is_last_enemy = i == wave.size() - 1
+func spawn_wave(wave) -> void:
+	for batch in wave:
+		print("[Enemy Spawn] Spawning batch ", batch)
+		for enemy_difficulty in batch:
+			var enemy = _spawn(enemy_difficulty)
+			enemy.died.connect(_on_enemy_died)
 
-		if is_last_enemy:
-			enemy.died.connect(wave_cleared.emit)
+		await get_tree().create_timer(1.0).timeout
 
-		await get_tree().create_timer(item.spawn_timer).timeout
 
-
-func is_valid_score(score: int) -> bool:
-	return ENEMIES.has(score)
+func _on_enemy_died() -> void:
+	if len(get_tree().get_nodes_in_group("enemies")) == 0:
+		wave_cleared.emit()
```

Comment:
may not be the best way but works well

noticed in the previous implementation, if you killed the last enemy (before the others) it would trigger the wave cleared signal, so just made it to where it didn't matter what order things were killed in.

Reply:
Since we take the enemies from the group on death, this should be great.